modules/miner.go

```diff
@@ -38,6 +48,10 @@ type Miner interface {
 	// indicating whether it has been solved, and an error.
 	SolveBlock(types.Block, types.Target) (types.Block, bool)
 
+	// GetWork() returns a MinerWork struct which can be converted to JSON to be
+	// parsed by external miners
+	GetWork() MinerWork
```

Comment:
Hmm. I'm not sure this is the best name, but I can't really think of a better one.

This should be returning a block header that the miner can grind on.